Lib/pyclbr.py

```diff
@@ -174,17 +198,22 @@ def _readmodule(module, path, inpackage=None):
                 tokentype, meth_name, start = next(g)[0:3]
                 if tokentype != NAME:
                     continue # Syntax error
+                cur_func = None
                 if stack:
-                    cur_class = stack[-1][0]
-                    if isinstance(cur_class, Class):
-                        # it's a method
-                        cur_class._addmethod(meth_name, lineno)
-                    # else it's a nested def
+                    cur_obj = stack[-1][0]
+                    if isinstance(cur_obj, Object):
```

Comment:
Thank you for clarifying that and for your patience.  I believe you're right -- the `isinstance(cur_obj, Object)` check isn't needed.  stack can only have Class or Function objects and cur_obj is only populated from the stack.